### ISSUE TITLE
Theme to get access to more artwork

### ISSUE THREAD

**Issue by wimpyrbx** [SickRage/SickRage#1456]
It would be nice to have templates being able to access more META artwork. I have modded my local sickrage copy to display background on DisplayShow template, but it's a dirty jquery/html hack to fix colors on tables etc since I am not 100% comfortable with the amount of code/css in the templates.

Live example here; http://www.fshare.it/image.jpeg?i=0q1460422rh

Jquery script for background used is called Backstretch.
Snippet i used in DisplayShow.mako:
`
$.backstretch('${srRoot}/showPoster/?show=${show.indexerid}&which=fanart');

$('.backstretch').css("opacity", 0.2).fadeIn("500");

`
* I also added style="opacity:0.8" to the information DIV and the main episode TABLE for image-effect.

So.. Also, a question; I see that SR downloads more META files than is available in the DATA/cache folder. I'm guessing they are just not saved into the cache folder but moved directly into the shows folder. It would be amazing to have more of these files to be saved to the cache folder.

Photoshopped image to show: http://www.fshare.it/image.jpeg?i=d41460469r9
1: banner, 2: season poster, 3: clearart, 4: show logo
Number 2,3 and 4 are photoshopped, the rest is already in place.

So if the above is possible in regards to allowing more types of artwork to be kept in cache + the showPoster routine be updated to allow more types (for example ?which=clearart) this theme can really come to life visually.

I will help as well as I can if someone wants to pick this up!

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by wimpyrbx** [comment]
if someone can start with helping me understand how to mod the showPoster routine to display more than fanart, banner and poster i can do some more testing locally by manually putting some files in place in DATA/cache for season poster, clearart etc.

I tried:
`${srRoot}/showPoster/?show=${show.indexerid}&which=**clearart**`
but that did obviously not work even tho i put an image in place with clearart in the name just like the other files are named.

**Comment by neoatomic** [comment]
Lets see if we can implement the fan-art backgrounds for the shows first. 
Hope @OmgImAlexis can take a look and see what can be done. :)

**Comment by wimpyrbx** [comment]
as a start for adding more artwork to the DisplayShow page.

`    <div id="container">
        <div id="posterCol">
            <a href="${srRoot}/showPoster/?show=${show.indexerid}&amp;which=poster" rel="dialog" title="${_('View Poster for')} ${show.name}"><img src="${srRoot}/showPoster/?show=${show.indexerid}&amp;which=poster_thumb" class="tvshowImg" alt=""/></a>
        </div>

        <div id="showCol">


<img src="${srRoot}/showPoster/?show=${show.indexerid}&amp;which=banner" style="float:right;height:50px;border:1px solid black">

            <div id="showinfo">
`

the img tag with banner is new here. it will display the shows banner to the right of the page like this:
http://www.fshare.it/image.jpeg?i=p614605364f

it should not interfere with any of the other code etc.

the background is easy to add:
`<script src='${srRoot}/js/backstretch.js'></script>
<script>
    $.backstretch('${srRoot}/showPoster/?show=${show.indexerid}&which=fanart');
    $('.backstretch').css("opacity", 0.2).fadeIn("500");
</script>`

however to make it "pretty" one would have to add opacity 0.8 for example to the summary box (to the right of the poster) and the main table. plus there is a need to remove the background color or change to transparent where it says "Season X". Like here:
http://www.fshare.it/image.jpeg?i=wb14605367p

so, fairly quick to fix and looks instantly more impressive.
getting these two things done is a great start.

next steps are as i see it;
* save more meta files (artwork) to cache folder (clearart, season posters, logo, landscape etc) when downloading artwork from the API
* allow ?show=${show.indexerid}&amp;which=poster to show more than poster, banner and fanart.

this will allow even more visual possibilities to increase the artwork usage in the templates :)

**Comment by OmgImAlexis** [comment]
@neoatomic this is backend stuff that needs to be changed. @wimpyrbx wants more types of artwork saved in the cache dir and other options added to the poster method to allow access from the front-end to other types of images instead of just the ones we have now.

**Comment by wimpyrbx** [comment]
Exactly OmgImAlexis
Once that is in place I can add some additions to the templates that perhaps can be included in the templates and turned on/off with a "template config setting"-page.

Display Banner [yes/no]
Display Clearart [yes/no]
etc etc.

**Comment by neoatomic** [comment]
@wimpyrbx Have been thinking a little about this.

The main concern i have is "over" downloading to much metadata/images. 
It would demand more disk storage, consume more bandwidth and maybe impact performance.  (Some of our users are running low powered/limited devices.) 
So i would be in favor to only download the essentials and store those by default in SR's own cache folder.

That being sad, there might be another way of doing things. 
Currently under the post-processing tab we have "metadata". That allows users to download/add season posters/banners etc. Those are stored in the shows own folder with the episode files.
What is we use those.? If /Game of thrones/season01-poster.jpg is present than show it on Displayshow.?
That will leave the user in control of the "download" and he can even replace it with others if he doesn't like a specific banner.

Just some thoughts, :)
